### DISCUSSION TITLE
CERN OHL V2

### DISCUSSION THREAD

**Discussion by cplpanic** [community/community#34150] [New to GitHub]
Hi everyone 

Getting confused on the difference between licence types and need some advice. 
Can anyone please explain the differences between .

weakly reciprocal
strongly reciprocal
permissive

https://choosealicense.com/non-software/

It's tricky to separate them and feel I'm missing context or just can't understand the legal jargon.
Looking to use this licence more in the future and could really do with some guidance. 

Don't mind people ,selling ,building, modifying just as long as an open version is kept intact. 

Sorry not very good with forums and unsure where to post.

**Comment by airtower-luna** [comment]
Broadly speaking, "permissive" usually means that permission to use the work comes with very few restrictions (often just keeping copyright notices intact and acknowledging there's no warranty). "Reciprocal" means that people using the work must use the same (or sometimes a similar) license for derived works. Mind that you *always* need to check the specific license for the exact conditions you have to follow. And if you need legal advice, talk to a copyright lawyer. :warning: 

At a glance, the difference between the "weakly reciprocal" and "strongly reciprocal" versions of the license you mentioned seems to be in section 4, how they deal with the covered work being included as *part* of a larger product. The strong one requires source for the full product to be made available, the weak one only for the parts based on what's under the license. That seems similar to the difference between GPL and LGPL.

**Reply by cplpanic** [reply]
Hi AirTower 

Thank you so much for answering, I was getting stressed trying to figure this out with very little background in law or development. 
So in essence reciprocal acts means a container...? If it's ok to use a wine or coke recipe as an example 

Strong = All changes to the recipe must stay within an agreed list. 

Weak = The original recipe are kept  but other varieties based upon it are fine. 

Permissive = Here is the recipe do what you want with it. 

**"talk to a copyright lawyer."**
I've try'd talking with local solicitors and lawyers but they where very confused about things such as open source \ copy left. 
Probably asked the wrong people.
 
After reading this a weak or permissive seems to fit but have heard people are against permissive licences..? 
Sorry just trying to quickly wrap my head around this under stress as there's another project that needs a licence and hardware is a nightmare. 

Thank you again feel much more grounded now :)  I in no way take help as legal advice and accept everything is gray and up-to myself in the end.

**Reply by airtower-luna** [reply]
Well, to stay with the drink example, let's assume you have a recipe for a drink under the different kinds of licenses.

* With a permissive license, you just have to mention where you got the recipe, and that any mistakes you made in mixing (or modifying the recipe) aren't the original author's fault.
* With the weakly reciprocal license, in addition to the former if you make a modified version of the drink you have to put your recipe under the same license.
* With the strongly reciprocal license, in addition to the former if you make a meal that includes the drink you have to put the recipe for the whole meal under the same license.

Reciprocal licenses are sometimes referred to as "copyleft" licenses.

Basically the difference is how strongly people who use the recipe are required to "give back" any changes or additions. That's also why some people prefer or dislike permissive licenses: Proponents of permissive licenses focus on giving the direct recipient maximum freedom. People who prefer reciprocal licenses tend to focus more on the common good created when people who work with the licensed code (or other work) have to share their work in turn.

Personally I'm leaning towards the latter view, but I'm not going to hate anyone for either choice. I tend to use copyleft licenses for complex code, and permissive ones for small stuff where I feel it doesn't matter that much what others do with it. :smile_cat: 

> I've try'd talking with local solicitors and lawyers but they where very confused about things such as open source \ copy left.
> Probably asked the wrong people.

Maybe? There's a lot of specialization between lawyers. :sweat_smile: There are some groups that look at legal matters around free software/open source (e.g. [SFLC](https://softwarefreedom.org/) or [FSFE](https://fsfe.org/activities/legal.en.html)). Maybe you can find some info there, probably depends on your jurisdiction too. :thinking: 

> Thank you again feel much more grounded now :) I in no way take help as legal advice and accept everything is gray and up-to myself in the end.

Excellent! :slightly_smiling_face:

**Reply by cplpanic** [reply]
Sorry for the delay - Got back and just wanted to sleep. 

Thank you so much for taking the time to help explain this to me would have been spinning around for weeks feeling very unsure - Feel alot more confident in using this licence now :) 

My opinion is towards the common good with some wiggle room so will probably start using the weak version on most projects, tho agree that permissive is good for light projects or if you made something and just want it out there. 

Again thank you (big hug)
